docs/managing_cr_status_sub_resources.md

```diff
@@ -0,0 +1,11 @@
+# Managing CR Status Sub Resources
+
+Operators manage systems on behalf of custom resources (CRs). The managed state
+of the system should always be reflected in the CRs status sub resource to make
+these information available for consumption. The recommended pattern here is to
+have [one operatorkit resource implementation to manage the system](https://github.com/giantswarm/chart-operator/tree/109f2b1f10d3500e4e9468ac901b0f344f37dd7b/service/controller/v3/resource/chart)
+and [one operatorkit resource implementation to provide the system's state in the CRs status sub resource](https://github.com/giantswarm/chart-operator/tree/109f2b1f10d3500e4e9468ac901b0f344f37dd7b/service/controller/v3/resource/chartstatus).
+In the linked example we have the `chart` resource and the `chartstatus`
+resource. The advantage of this pattern is to have dedicated operatorkit
+resource implementations which can reconcile independently. That way different
+resync periods can be used for each operatorkit resource implementation.
```

Comment:
I can see that not always possible/convenient. Let's say we want to put node IP in the status. The node is created by CF stack, and the information about the node is obtained from CF outputs. Then the most convenient way to set node IP in the status is doing that from the same resource.

In general if status information is obtained from the API that makes actual changes it's convenient to update the status immediately after getting the API result.

I can agree with this as a general guideline though if that's the intention.

Reply:
I see what you mean. I also think this is not black and white. Best practises with the status management will also evolve further as we go. I just wanted to have a first document and start describing what we think. One aspect which is not really clear to me is who reads and writes status and in which combination does this occur to be relevant or dramatically different. As a guideline I would vote to always try to have the separated resources though because it keeps different responsibilities separated so stuff does not get mixed up. 

Anything to rewrite or add here now to make you happy?

Reply:
I added some notes about the default approach and the evolving nature and complexity of the status management.